Followup of #15771

vipulnsward opened the pull request:
Make sure we handle explicitly passed nil's to lock_version as well

Fixes #24695

r? @sgrif

maclover7 commented:
Nice use of the new attributes API 👍

sgrif commented on activerecord/lib/active_record/locking/optimistic.rb:
```diff
@@ -190,6 +190,11 @@ def deserialize(value)
       end
 
+      def serialize(value)
+        # `nil` *should* be changed to 0
```
Can we kill this comment, and mention why `nil` should be changed to `0` in the commit message instead?

jeremy replied:
This is when we want a comment on the _behavior of the code_, not just a comment on the _intent of the commit_.

We have some code that intentionally explicitly coerces `nil` to an integer, and it's code that looks like it may be an inadvertent bug. A comment _on the code_ clarifies that this is desired behavior. The comment lives with the behavior, not just with the commit that introduced the behavior.

vipulnsward replied:
A similar comment still exists on deserialize. Should that suffice, or it needs to be added here as well?

jeremy replied:
Could discuss it in a comment on LockingType instead.

vipulnsward committed:
Followup of #15771

Make sure we handle explicitly passed nil's to lock_version as well.
An explicitly passed nil value is now converted to 0 on LockingType,
so that we don't end up with ActiveRecord::StaleObjectError in update record
optimistic locking

Fixes #24695

vipulnsward commented on activerecord/lib/active_record/locking/optimistic.rb:
```diff
@@ -190,6 +190,10 @@ def deserialize(value)
 
+      def serialize(value)
+        super.to_i
+      end
```
removed
